README.md

```diff
@@ -2,5 +2,435 @@
+    clock: Clock
+) : UnitOfWork<ServicePrincipal, Params, Wallet>(clock) {
+
+    @Serializable
+    data class Params(val id: String, val currency: String) : UowParams<Params> {
+        override fun serialization() = serializer()
+    }
+
+    override suspend fun tryPerform(principal: ServicePrincipal, params: Params): Changes<Wallet> {
+        val walletId = Wallet.Id(UUID.fromString(params.id))
+        val wallet = queries.find(walletId)
+
+        return if (wallet != null) {
+            noChanges(wallet)
+        } else {
+            val amount = ULong.MIN_VALUE
+            val currency = Currency.getInstance(params.currency)
+            val expireAt = clock.instant().plus(timeToExpire)
+            val newWallet = Wallet(
+                id = walletId,
+                currency = currency,
+                amount = amount,
+                expireAt = expireAt,
+                entityState = newState(WalletEvent.Created(walletId, currency, amount, expireAt))
+            )
+            changes {
+                add(newWallet)
+            }
+        }
+    }
+    companion object {
+        private val timeToExpire = Duration.ofDays(600)
+    }
+}
+```
+In this example we return noChanges in case model with such id already exists, otherwise we create new model and using ChangesDsl we **add** new model.
+
+
+You can also update your models in scope of unit of work - 
+```kotlin
+    changes {
+        update(wallet.deposit(amount))    
+    }
+```
+By default, **update** allows passing model with no changes (model was not updated after calling *deposit()* method).
+Sometimes it can lead to some inconsistency in your domain logic - you expected model to be changed, but it wasn't.
+You can force verification for your model, that it was changed in scope of your unit of work:
+```kotlin
+    changes {
+        update(wallet.deposit(amount), required = true)    
+    }
+```
+
+### Repository
+To persist our model we need to add repository for it.
+We use [jOOQ](https://www.jooq.org/) to have a type-safe DB querying.
+You need to generate jOOQ tables/records based on your DB schema to have a type-safe mapping of your model to DB record.
+You can use different Gradle plugins to generate jOOQ tables, f.e. check this [plugin](https://github.com/etiennestuder/gradle-jooq-plugin).
+
+Your generated records should extend [BaseEntityRecord](eva-jooq/src/main/kotlin/com/razz/jooq/record/BaseEntityRecord.kt).
+To achieve it use [jOOQ matcher strategies](https://www.jooq.org/doc/latest/manual/code-generation/codegen-matcherstrategy/).
+
+When you create tables for your models you need to add next fields to your schema, so we can persist your model properly - 
+```sql
+  record_updated_at         TIMESTAMP      NOT NULL            ,
+  record_created_at         TIMESTAMP      NOT NULL            ,
+  version                   BIGINT         NOT NULL
+```
+
+After you created DB schema for you data, we can implement Repository for your model.
```

Comment:
why someone referred as _you_ have to create schema, but _we_ are the ones who "can implement Repository"?